File: panel/backend/app/api/tools.py
from fastapi import APIRouter
from app.schemas import Target, Domain, Command
from app.services import run_in_container, list_reports, get_report

router = APIRouter(prefix="/api", tags=["tools"])


@router.post("/nmap")
def nmap_scan(payload: Target):
    return run_in_container("nmap-suite", ["nmap", "-sV", payload.target])


@router.post("/recon")
def recon_scan(payload: Domain):
    return run_in_container("recon", ["subfinder", "-d", payload.domain])


@router.post("/hackagent")
def hackagent(payload: Target):
    return run_in_container("hackagent", ["hackagent", "--help"])


@router.post("/autopentestx")
def autopentestx(payload: Target):
    return run_in_container(
        "autopentestx", ["python3", "/app/main.py", "--version"]
    )


@router.post("/inspector")
def inspector(payload: Target):
    return run_in_container(
        "inspector", ["python3", "/app/core/inspector.py", "-h"]
    )


@router.post("/burp")
def burp(payload: Command):
    return run_in_container("burp", ["bash", "-lc", payload.command])


@router.post("/kali")
def kali(payload: Command):
    return run_in_container("kali-tools", ["bash", "-lc", payload.command])


@router.get("/reports")
def reports():
    return {"reports": list_reports()}


@router.get("/reports/{report_id}")
def report_details(report_id: str):
    return get_report(report_id)

File: panel/frontend/src/lib/services/tools.ts
import { api } from '$lib/api';
import type { ToolOutput, ReportsResponse, StatusResponse } from '$lib/types';

export function nmapScan(target: string) {
  return api<ToolOutput>('/api/nmap', 'POST', { target });
}

export function reconScan(domain: string) {
  return api<ToolOutput>('/api/recon', 'POST', { domain });
}

export function hackAgent(target: string) {
  return api<ToolOutput>('/api/hackagent', 'POST', { target });
}

export function autopentestX(target: string) {
  return api<ToolOutput>('/api/autopentestx', 'POST', { target });
}

export function inspector(target: string) {
  return api<ToolOutput>('/api/inspector', 'POST', { target });
}

export function burp(command: string) {
  return api<ToolOutput>('/api/burp', 'POST', { command });
}

export function kali(command: string) {
  return api<ToolOutput>('/api/kali', 'POST', { command });
}

export function getStatus() {
  return api<StatusResponse>('/api/status', 'GET');
}

export function getReports() {
  return api<ReportsResponse>('/api/reports', 'GET');
}
